covariance/covarianceXsec.cpp

```diff
@@ -571,133 +746,179 @@ void covarianceXsec::scanParameters() {
-      }//End Near affecting normpars
-      else if (GetXSecParamID(i, 0) == -2) {//Near affecting func pars
-	
-        NearfuncParsNames.push_back(GetParameterName(i));
-        NearfuncParsIndex.push_back(i);
-        nNearFuncParams++;
-	
-        // If they are greater >= 0 it's a spline parameter
-      }//End Near affecting func pars
-      else if (GetXSecParamID(i, 0) >= 0) {
-        NearsplineParsNames.push_back(GetParameterName(i));
-        NearsplineParsIndex.push_back(i);
-        nNearSplineParams++;
-
-        if(xsec_spline_interpolation != NULL)
-        {
-          for(int il = 0; il < kSplineInterpolations; ++il)
-          {
-            if (std::string(((TObjString*)xsec_spline_interpolation->At(i))->GetString()) == SplineInterpolation_ToString(SplineInterpolation(il)))
-              SplineInterpolationType.push_back(SplineInterpolation(il));
-          }
-          if (SplineInterpolationType.size() != static_cast<unsigned int>(nNearSplineParams))
-          {
-            std::cerr<<" "<<"Couldn't find interpoaltion for paramter " <<nPars<< " it has interpoaltion "<<std::string(((TObjString*)xsec_spline_interpolation->At(i))->GetString())<<std::endl;
-            throw;
-          }
-        }
-	//Fill the name of the Far spline objects in the spline files
-	//NearSplineFileParsNames.push_back(std::string(((TObjString*)xsec_param_nd_spline_names->At(i))->GetString()));
-      }//End Near affecting spline pars
-      else {
-	std::cerr << "Found a parameter in covarianceXsec which wasn't -2, -1 or above 0!" << std::endl;
-	std::cerr << "This is undefined behaviour currently, and implementation should change" << std::endl;
-	std::cerr << "Param " << GetParameterName(i) << " (param " << i << ") = " << GetXSecParamID(i, 0) << std::endl;
-	std::cerr << __LINE__ << ":" << __LINE__ << std::endl;
-	throw;
-      }
-    }//End Near affecting parameters
+		//Next ones are kinematic bounds on where normalisation parameter should apply (at the moment only etrue but hope to add q2
+		//We set a bool to see if any bounds exist so we can shortcircuit checking all of them every step
+		bool haskinbounds=false;
+
+		////////////////////
+		//New generic cuts things 
+		////////////////////
+
+		//Only consider the kinematic string and the boundaries if you've actually given it a string to use...
+		if( _fKinematicPars[i].size() > 0 ){
+		  haskinbounds = true;
+
+		  //ETA - This can be a vector :) can provide different kinematic variables to cut on
+		  tmp_xsec.KinematicVarStr = _fKinematicPars[i];
+
+		  //ETA - This can be a vector :) can provide different kinematic variables to cut on
+		  std::vector< std::vector<double> > Selections(_fKinematicPars[i].size());
+
+		  //ETA - push back kinematic type with dummy -999 since this needs to be converted into an enum for a kinematic type within
+		  //a samplePDFFD daughter class
+		  for(int KinVar_i = 0 ; KinVar_i < _fKinematicPars[i].size() ; ++KinVar_i) {
```

Comment:
Super nitpicky but can you cast _fKinematicPars[i].size() to an integer (or KinVar_i to size_t)